packages/keyring-api/src/api/transaction.ts

```diff
@@ -0,0 +1,323 @@
+ *   timestamp: 1716367781,
+ *   type: 'send',
+ *   from: [
+ *     {
+ *       address: 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ *   to: [
+ *     {
+ *       address: 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *     {
+ *       address: 'bc1qwl8399fz829uqvqly9tcatgrgtwp3udnhxfq4k',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ *   fees: [
+ *     {
+ *       type: 'priority',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ * };
+ * ```
+ */
+export const TransactionStruct = object({
+  /**
+   * Chain-specific transaction ID.
+   */
+  id: string(),
+
+  /**
+   * Chain ID (CAIP-2).
+   */
+  chain: CaipChainIdStruct,
+
+  /**
+   * Account ID (UUIDv4).
+   */
+  account: UuidStruct,
+
+  /**
+   * Transaction status {@see TransactionStatus}.
+   */
+  status: enums([
+    `${TransactionStatus.Submitted}`,
+    `${TransactionStatus.Unconfirmed}`,
+    `${TransactionStatus.Confirmed}`,
+    `${TransactionStatus.Failed}`,
+  ]),
+
+  /**
+   * UNIX timestamp of when the transaction was added to the blockchain. The
+   * timestamp can be null if the transaction has not been included in the
+   * blockchain yet.
+   */
+  timestamp: nullable(number()),
+
+  /**
+   * Transaction type {@see TransactionType}. This will be used by MetaMask to enrich the transaction
+   * details on the UI.
+   */
+  type: enums([`${TransactionType.Send}`, `${TransactionType.Receive}`]),
+
+  /**
+   * Transaction sender addresses and amounts.
+   */
+  from: array(ParticipantStruct),
+
+  /**
+   * Transaction receiver addresses and amounts.
+   */
+  to: array(ParticipantStruct),
+
+  /**
+   * Total transaction fee.
+   */
+  fees: array(FeeStruct),
+
+  /**
+   * List of events related to the transaction {@see TransactionEventStruct}.
+   *
+   * The events are tracked in a best-effort basis and may not be available for
+   * all transactions.
+   */
+  events: array(TransactionEventStruct),
+});
+
+/**
+ * Transaction object.
+ *
+ * See {@link TransactionStruct}.
+ */
+export type Transaction = Infer<typeof TransactionStruct>;
+
+/**
+ * This struct represents a page of transactions.
+ *
+ * @example
+ * ```ts
+ * {
+ *   data: [
+ *     {
+ *       // Transaction object
+ *     }
+ *   ],
+ *   next: 'c3y1Q6QtqtstbxKX+oqVdEW6',
+ * }
+ * ```
+ */
```

Comment:
```suggestion
 *
 * @example
 * ```ts
 * {
 *   data: [
 *     {
 *       // Transaction object
 *     }
 *   ],
 *   next: null, // No more results
 * }**
 * ```
 */
```